byway/src/main/webapp/track-interests.js

```diff
@@ -29,42 +30,53 @@ if(document.readyState === 'loading') {
  * they can select.
  */
 function loadContent() {
-  configureTripKey();
-  loadButtonsWithInterests();
-  setupLogoutLink();
-}
-
-/* Gets the trip key from the url and sets it for the next page. */
-function configureTripKey() {
   let nextPage = document.getElementById("next-button");
-  tripKey = getTripKeyFromUrl();
   nextPage.href = configureTripKeyForPath(tripKey, "/routepage.html");
+  getTripInterests();
+  loadButtonsWithInterests();
+  setupLogoutLink();
 }
 
 /** 
- * Check if a button's value is stored in the set interestsChosen.
+ * Check if a button's value is stored in the set newInterests.
  * Remove from the set and remove the active class if the interest
  * was previously chosen and add if the interest was not included.
  * @param {String} place text value of the button's interest
  * @param {Element} elem tracks the current button element chosen.
  */
 function updateStatus(place, elem) {
-  if(interestsChosen.has(place)) {
-    interestsChosen.delete(place);
+  if(newInterests.has(place)) {
+    newInterests.delete(place);
     elem.classList.remove('active');
   } else {
-    interestsChosen.add(place);
+    newInterests.add(place);
     elem.classList.add('active');
   }
 }
 
+/**
+ * Finds the list of interests that were set for the trip.
+ * If it is a new trip, no interests are listed. If the trip
+ * had previously-saved interests, return them.
+ * @return fulfilled promise containing the list of interests
+ */
+async function getTripInterests() {
+  await fetchInterests(tripKey)
+  .then((interests) => {
+    for (let interest of interests) {
+      interestsSaved.add(interest);
+    }
+  });
```

Comment:
Try to avoid mixing async/await with promise-chaining. Both do the same thing, so just use one or the other.
Reminder these do the same thing:

```js
function getTripInterests() {
  return fetchInterests(tripKey)
    .then((interests) => {
      for (let interest of interests) {
        interestsSaved.add(interest);
      }
    });
}

async function getTripInterests() {
  let interests = await fetchInterests(tripKey);
  for (let interest of interests) {
    interestsSaved.add(interest);
  }
}
```

Reply:
Fixed